Generics: the trait bound `Value: ToSchema` is not satisfied

Hey!

When using a generic as the `body` property in path macro I'm getting two errors:

- error[E0277]: the trait bound `Value: ToSchema` is not satisfied
- error[E0277]: the trait bound `Value: ComposeSchema` is not satisfied

Code (the relevant parts):

```rust
    use std::net::{Ipv4Addr, SocketAddr};
    
    use std::io::Error;
    use tokio::net::TcpListener;
    use utoipa::{openapi::security::{ApiKey, ApiKeyValue, SecurityScheme}, Modify, OpenApi};
    use utoipa_axum::router::OpenApiRouter;
    use utoipa_redoc::{Redoc, Servable};

    #[derive(Debug, Clone, Default, Serialize, ToSchema, Deserialize)]
    pub struct ResponseSchema<D, M = Option<serde_json::Value>> {
        pub data_one: D,
        #[schema(value_type=Object)]
        pub data_two: M,
    }

    #[derive(Serialize, Deserialize, ToSchema, Clone)]
    struct Todo {
        id: i32,
        #[schema(example = "Buy groceries")]
        value: String,
        done: bool,
    }

    #[utoipa::path(
        get,
        path = "",
        tag = TODO_TAG,
        responses(
            (status = 200, description = "List all todos successfully", body = [ResponseSchema<Todo>])
        )
    )]
    async fn list_todos(State(store): State<Arc<Store>>) -> Json<Vec<Todo>> {
        let todos = store.lock().await.clone();

        Json(todos)
    }
```

What is weird is that i swear it was working a few hours ago and it started after I updated the crate to 5.0.0. But I don't have my previous lock file to compare what changed.

I've ran cargo clean and tried in a new project as well.


<details>
  <summary>Full code</summary>
  ```

    use std::net::{Ipv4Addr, SocketAddr};
    
    use std::io::Error;
    use tokio::net::TcpListener;
    use utoipa::{openapi::security::{ApiKey, ApiKeyValue, SecurityScheme}, Modify, OpenApi};
    use utoipa_axum::router::OpenApiRouter;
    use utoipa_redoc::{Redoc, Servable};
    
    const TODO_TAG: &str = "todo";
    
    #[tokio::main]
    async fn main() -> Result<(), Error> {
        #[derive(OpenApi)]
        #[openapi(
            modifiers(&SecurityAddon),
            tags(
                (name = TODO_TAG, description = "Todo items management API")
            )
        )]
        struct ApiDoc;
    
        struct SecurityAddon;
    
        impl Modify for SecurityAddon {
            fn modify(&self, openapi: &mut utoipa::openapi::OpenApi) {
                if let Some(components) = openapi.components.as_mut() {
                    components.add_security_scheme(
                        "api_key",
                        SecurityScheme::ApiKey(ApiKey::Header(ApiKeyValue::new("todo_apikey"))),
                    )
                }
            }
        }
    
        let (router, api) = OpenApiRouter::with_openapi(ApiDoc::openapi())
            .nest("/api/v1/todos", todo::router())
            .split_for_parts();
    
        let router = router
            .merge(Redoc::with_url("/redoc", api.clone()));
    
        let address = SocketAddr::from((Ipv4Addr::UNSPECIFIED, 8080));
        let listener = TcpListener::bind(&address).await?;
        axum::serve(listener, router.into_make_service()).await
    }
    
    mod todo {
        use std::sync::Arc;
    
        use axum::{
            extract::{State},
            Json,
        };
        use serde::{Deserialize, Serialize};
        use tokio::sync::Mutex;
        use utoipa::{ToSchema};
        use utoipa_axum::{router::OpenApiRouter, routes};
    
        #[derive(Debug, Clone, Default, Serialize, ToSchema, Deserialize)]
        pub struct ResponseSchema<D, M = Option<serde_json::Value>> {
            pub data_one: D,
            #[schema(value_type=Object)]
            pub data_two: M,
        }
    
        use crate::TODO_TAG;
    
        /// In-memory todo store
        type Store = Mutex<Vec<Todo>>;
    
        /// Item to do.
        #[derive(Serialize, Deserialize, ToSchema, Clone)]
        struct Todo {
            id: i32,
            #[schema(example = "Buy groceries")]
            value: String,
            done: bool,
        }
    
        pub(super) fn router() -> OpenApiRouter {
            let store = Arc::new(Store::default());
            OpenApiRouter::new()
                .routes(routes!(list_todos))
                .with_state(store)
        }
    
        /// List all Todo items
        ///
        /// List all Todo items from in-memory storage.
        #[utoipa::path(
            get,
            path = "",
            tag = TODO_TAG,
            responses(
                (status = 200, description = "List all todos successfully", body = [ResponseSchema<Todo>])
            )
        )]
        async fn list_todos(State(store): State<Arc<Store>>) -> Json<Vec<Todo>> {
            let todos = store.lock().await.clone();
    
            Json(todos)
        }
    }
    ```
</details>

--- a/utoipa/CHANGELOG.md
+++ b/utoipa/CHANGELOG.md
@@ -8,6 +8,7 @@ to look into changes introduced to **`utoipa-gen`**.
 ### Added
 
 * Implement schema traits for indexset (https://github.com/juhaku/utoipa/pull/1129)
+* Add `ToSchema` implementation for serde_json::Value (https://github.com/juhaku/utoipa/pull/1132)
 
 ## 5.0.0 - Oct 14 2024
 

--- a/utoipa/src/lib.rs
+++ b/utoipa/src/lib.rs
@@ -779,6 +779,16 @@ where
     }
 }
 
+impl PartialSchema for serde_json::Value {
+    fn schema() -> openapi::RefOr<openapi::schema::Schema> {
+        utoipa::openapi::schema::Object::builder()
+            .schema_type(utoipa::openapi::schema::SchemaType::AnyValue)
+            .into()
+    }
+}
+
+impl ToSchema for serde_json::Value {}
+
 // Create `utoipa` module so we can use `utoipa-gen` directly from `utoipa` crate.
 // ONLY for internal use!
 #[doc(hidden)]

